frontend/src/components/widgets/DataFrame/hooks/useColumnLoader.ts

```diff
@@ -38,17 +38,29 @@ export const INDEX_IDENTIFIER = "index"
 // Prefix used in the config column mapping when referring to a column via the numeric position
 export const COLUMN_POSITION_PREFIX = "col:"
 
+export const COLUMN_WIDTH_MAPPING = {
+  small: 75,
+  medium: 200,
+  large: 400,
+}
+
 /**
  * Options to configure columns.
+ *
+ * This needs to be kept in sync with the ColumnConfig TypeDict in the backend.
+ * This will eventually replaced with a proto message.
  */
 export interface ColumnConfigProps {
-  width?: number
   title?: string
-  type?: string
+  width?: "small" | "medium" | "large"
   hidden?: boolean
-  editable?: boolean
-  metadata?: Record<string, unknown>
-  alignment?: string
+  disabled?: boolean
+  required?: boolean
+  default?: number | string | boolean
+  type?: string
+  // uses snake_case to match the property names in the backend:
+  type_options?: Record<string, unknown>
+  alignment?: "left" | "center" | "right"
```

Comment:
nit / question: Should this be an enum?

Reply:
Since this data structure is just for matching with the loaded column config JSON structure and converting it into the internal column props, it probably won't work with an enum or not be really useful to have an enum here. But I will have a look if an enum would make sense for the internal column props.